lib/rabbitmqctl.ex

```diff
@@ -67,13 +67,75 @@ defmodule RabbitMQCtl do
       _ ->
         options = parsed_options |> merge_all_defaults |> normalize_options
         {arguments, options} = command.merge_defaults(arguments, options)
+
         with_distribution(options, fn() ->
-          validate_and_run_command(options, command, arguments)
-          |> handle_command_output(command, options, unparsed_command, output_fun)
+          # The code below implements a tiny decision tree that has
+          # to do with CLI argument and environment state validation.
+
+          # validate CLI arguments
+          case command.validate(arguments, options) do
+            :ok ->
+              # then validate RabbitMQ application state (running or not)
+              case maybe_validate_rabbit_app_state(command, arguments, options) do
+                :ok ->
+                  # then optionally validate execution environment
+                  case maybe_validate_execution_environment(command, arguments, options) do
+                    :ok   -> proceed_to_execution(command, arguments, options)
+                    other -> other
+                  end
+                other -> other
+              end
+            other -> other
+          # handle_command_output will handle all the errors
+          # among other things
+          end |> handle_command_output(command, options, unparsed_command, output_fun)
         end)
     end
   end
 
+  defp maybe_validate_rabbit_app_state(_command, _arguments, %{offline: true}) do
+    :ok
+  end
+  defp maybe_validate_rabbit_app_state(command, arguments, options) do
+    required_state = case function_exported?(command, :required_rabbit_app_state, 2) do
+                       false -> :running
```

Comment:
Adding a validator which does nothing just to avoid this default validation is even more annoying. Because it will not tell anything about the command, but work around the system defaults.